.github/workflows/patchTemplate.yml

```diff
@@ -0,0 +1,93 @@
+name: Patch Template
+
+on:
+  pull_request:
+    types: [closed]
+    branches:
+      - main
+
+jobs:
+  template-matrix:
+    runs-on: ubuntu-latest
+    steps:
+      - id: templates
+        run: |
+          framework=`echo '${{ github.head_ref }}' | jq -Rr 'split("-")[2]'`;
+          templates=$(gh search repos --owner searchspring "snapfu-template-$framework" --json name --jq '[.[].name]')
+          echo "templates=$templates" >> $GITHUB_OUTPUT
+        env:
+          GITHUB_TOKEN: ${{ secrets.MACHINE_ACTION_WORKFLOW_PAT }}
+    outputs:
+      templates: ${{ steps.templates.outputs.templates }}
+  patch-template:
+    if: github.event.pull_request.merged == true && startsWith( github.head_ref, 'patch-generate-' ) == true && needs.template-matrix.outputs.templates != '[]'
+    runs-on: ubuntu-latest
+    needs:
+      - template-matrix
+    strategy:
+      matrix:
+        template: ${{ fromJSON(needs.template-matrix.outputs.templates) }}
+    steps:
+      - name: Extract Variables
+        id: variables
+        run: |
+          framework=`echo '${{ github.head_ref }}' | jq -Rr 'split("-")[2]'`;
+          echo "::set-output name=framework::$framework";
+          echo "Using framework: $framework";
+
+          version=`echo '${{ github.head_ref }}' | jq -Rr 'split("-")[3]'`;
+          echo "::set-output name=version::$version";
+          echo "Using version: $version";
+
+      - name: Validate Variables
+        run: |
+          if [ "${{ steps.variables.outputs.framework }}" == "" ]; then 
+            echo "Could not determine framework";
+            exit 1;
+          fi;
+          if [ "${{ steps.variables.outputs.version }}" == "" ]; then 
+            echo "Could not determine version";
+            exit 1;
+          fi;
+
+      - name: Setup Node
+        id: setup-node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+
+      - name: Install Snapfu
+        run: |
+          npm install -g snapfu
+
```

Comment:
Can this be done outside of the matrix loop to prevent it from happening over and over unnecessarily?

Reply:
Looks like both the 'needs' dependancy and each entry of the matrix runs in a separate environment so I'm not seeing a way.

Reply:
Alright - just thought it would be good to check :)